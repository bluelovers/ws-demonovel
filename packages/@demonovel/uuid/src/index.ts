/**
 * Created by user on 2020/6/14.
 */

import { v5 as uuidv5 } from 'uuid';

export function newNovelUUID(siteID: string, uniqueID: string | number): string
{
	let seed = siteID + '#' + String(uniqueID);
	//return hashSum(seed)
	return uuidv5(seed, uuidv5.URL)
}

export default newNovelUUID
