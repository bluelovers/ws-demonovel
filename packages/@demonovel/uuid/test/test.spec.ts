//@noUnusedParameters:false
/// <reference types="jest" />
/// <reference types="node" />
/// <reference types="expect" />

import { basename, extname } from 'path';
import { newNovelUUID } from '../src/index';

beforeAll(async () =>
{

});

describe(basename(__filename, extname(__filename)), () =>
{

	test.skip(`dummy`, () => {});

	test(`test`, () =>
	{

		let actual = newNovelUUID('demonovel', 'user2/其實，我乃最強');
		expect(actual).toMatchSnapshot();

	});

})
