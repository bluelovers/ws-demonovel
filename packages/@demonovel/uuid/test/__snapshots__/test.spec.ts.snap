// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test.spec test 1`] = `"17042753-548e-5f12-9f7d-afdcb4c15a6e"`;
